onnxruntime/core/protobuf/onnx-ml.proto

```diff
@@ -633,4 +762,15 @@ message FunctionProto {
   repeated NodeProto node = 7;
   // A human-readable documentation for this function. Markdown is allowed.
   optional string doc_string = 8;
+  
+  // The OperatorSets this function body (graph) relies on.
+  // A FunctionProto body (graph) may implicitly rely on the OperatorSet that
+  // this function belongs to. It can also explicitly rely on more OperatorSets
+  // with this field specified.
+  //
+  // All nodes in the function body (graph) will bind against the operator
+  // with the same-domain/same-op_type operator with the HIGHEST version
+  // in the referenced operator sets. This means at most one version can be relied
+  // for one domain.
+  repeated OperatorSetIdProto opset_import = 9;
 }
```

Comment:
This update is a bit deceptive as in the original onnx files FunctionProto moved to onnx-operators.in.proto. This becomes a far more significant diff over onnx, purely to maintain the custom addition a repeated FunctionProto field in the Model. 

Does the training code need that facility? Currently the only usage is one unit test in the ORT code. 

If we're going to have this big diff, the documentation in docs\How_To_Update_ONNX_Dev_Notes.md for updating the ONNX version needs updating to call it out and mention that the merge is now more manual and you need to apply any changes to FunctionProto from the ONNX version to onnx-operators.in.proto to the ORT version of onnx.in.proto.